### ISSUE TITLE
TypeError: 'NoneType' object is not subscriptable

### ISSUE THREAD

**Issue by TrueWinter** [akpw/mktxp#11]
I was successfully able to get `mktxp` working with a RouterBoard 951 (single WiFi interface and DHCP server). But crashes when trying to set it up with a GrooveA 52 (3 WiFi interfaces, 2 DHCP servers, 1 hotspot). The crash log has been included below:

```
$ mktxp print -en outdoor-ap -wc
Connecting to router outdoor-ap@192.168.1.158
2021-07-15 21:44:21 Connection to router outdoor-ap@192.168.1.158 has been established
Traceback (most recent call last):
  File "/usr/local/bin/mktxp", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/dist-packages/mktxp/cli/dispatch.py", line 106, in main
    MKTXPDispatcher().dispatch()
  File "/usr/local/lib/python3.6/dist-packages/mktxp/cli/dispatch.py", line 42, in dispatch
    self.print(args)
  File "/usr/local/lib/python3.6/dist-packages/mktxp/cli/dispatch.py", line 96, in print
    OutputProcessor.wifi_clients(args['entry_name'])
  File "/usr/local/lib/python3.6/dist-packages/mktxp/flow/processor/base_proc.py", line 60, in wifi_clients
    WirelessOutput.clients_summary(router_entry)
  File "/usr/local/lib/python3.6/dist-packages/mktxp/cli/output/wifi_out.py", line 37, in clients_summary
    BaseOutputProcessor.augment_record(router_entry, registration_record, dhcp_lease_records)
  File "/usr/local/lib/python3.6/dist-packages/mktxp/flow/processor/output.py", line 55, in augment_record
    registration_record['rx_rate'] = BaseOutputProcessor.parse_rates(registration_record['rx_rate'])
  File "/usr/local/lib/python3.6/dist-packages/mktxp/flow/processor/output.py", line 88, in parse_rates
    return f'{int(float(rc[1]))} {rc[2]}'
TypeError: 'NoneType' object is not subscriptable
```

The error does not occur with the `mktxp print -en outdoor-ap -dc` command.

**Comment by akpw** [comment]
@TrueWinter thanks for the bug report, will fix in next version

**Comment by akpw** [comment]
@TrueWinter thanks for a bug report, will fix in next version

**Comment by eblet** [comment]
@akpw 
```
Exception happened during processing of request from ('127.0.0.1', 34578)
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/socketserver.py", line 320, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/local/lib/python3.6/socketserver.py", line 351, in process_request
    self.finish_request(request, client_address)
  File "/usr/local/lib/python3.6/socketserver.py", line 364, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/local/lib/python3.6/socketserver.py", line 724, in __init__
    self.handle()
  File "/usr/local/lib/python3.6/http/server.py", line 418, in handle
    self.handle_one_request()
  File "/usr/local/lib/python3.6/http/server.py", line 406, in handle_one_request
    method()
  File "/usr/local/lib/python3.6/site-packages/prometheus_client/exposition.py", line 240, in do_GET
    status, header, output = _bake_output(registry, accept_header, params)
  File "/usr/local/lib/python3.6/site-packages/prometheus_client/exposition.py", line 105, in _bake_output
    output = encoder(registry)
  File "/usr/local/lib/python3.6/site-packages/prometheus_client/openmetrics/exposition.py", line 27, in generate_latest
    for k, v in sorted(s.labels.items())]))
  File "/usr/local/lib/python3.6/site-packages/prometheus_client/openmetrics/exposition.py", line 27, in <listcomp>
    for k, v in sorted(s.labels.items())]))
AttributeError: ("'NoneType' object has no attribute 'replace'", Metric(mktxp_wlan_registered_clients, Number of registered clients, gauge, , [Sample(name='mktxp_wlan_registered_clients', labels={'channel': '2452/20-Ce/gn(17dBm)', 'routerboard_name': 'AM09-MT01', 'routerboard_address': '192.168.81.1'}, value='1', timestamp=None, exemplar=None), Sample(name='mktxp_wlan_registered_clients', labels={'channel': '5640/20-eCee/ac/DP(24dBm)', 'routerboard_name': 'AM09-MT01', 'routerboard_address': '192.168.81.1'}, value='1', timestamp=None, exemplar=None), Sample(name='mktxp_wlan_registered_clients', labels={'channel': None, 'routerboard_name': 'AM09-MT01', 'routerboard_address': '192.168.81.1'}, value='1', timestamp=None, exemplar=None)]))
```
Does this error fit into this issue?

**Comment by akpw** [comment]
@TrueWinter , this should be fixed in the latest

**Comment by TrueWinter** [comment]
The `mktxp print -en outdoor-ap -wc` command does now work, however running `mktxp export` and accessing the port returns the following error:

```
Exception happened during processing of request from ('127.0.0.1', 49182)
Traceback (most recent call last):
  File "/usr/lib/python3.6/socketserver.py", line 320, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/lib/python3.6/socketserver.py", line 351, in process_request
    self.finish_request(request, client_address)
  File "/usr/lib/python3.6/socketserver.py", line 364, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python3.6/socketserver.py", line 724, in __init__
    self.handle()
  File "/usr/lib/python3.6/http/server.py", line 418, in handle
    self.handle_one_request()
  File "/usr/lib/python3.6/http/server.py", line 406, in handle_one_request
    method()
  File "/home/nicholis/.local/lib/python3.6/site-packages/prometheus_client/exposition.py", line 278, in do_GET
    status, headers, output = _bake_output(registry, accept_header, accept_encoding_header, params, False)
  File "/home/nicholis/.local/lib/python3.6/site-packages/prometheus_client/exposition.py", line 104, in _bake_output
    output = encoder(registry)
  File "/home/nicholis/.local/lib/python3.6/site-packages/prometheus_client/openmetrics/exposition.py", line 34, in generate_latest
    for k, v in sorted(s.labels.items())]))
  File "/home/nicholis/.local/lib/python3.6/site-packages/prometheus_client/openmetrics/exposition.py", line 34, in <listcomp>
    for k, v in sorted(s.labels.items())]))
AttributeError: ("'NoneType' object has no attribute 'replace'", Metric(mktxp_wlan_registered_clients, Number of registered clients, gauge, , [Sample(name='mktxp_wlan_registered_clients', labels={'channel': '2462/20/gn', 'routerboard_name': 'outdoor-ap', 'routerboard_address': '192.168.1.158'}, value='8', timestamp=None, exemplar=None), Sample(name='mktxp_wlan_registered_clients', labels={'channel': None, 'routerboard_name': 'outdoor-ap', 'routerboard_address': '192.168.1.158'}, value='2', timestamp=None, exemplar=None), Sample(name='mktxp_wlan_registered_clients', labels={'channel': None, 'routerboard_name': 'outdoor-ap', 'routerboard_address': '192.168.1.158'}, value='2', timestamp=None, exemplar=None)]))
```

Here's the config file, if it will help in debugging:
```
[rb951]
    route = True
    use_comments_over_names = True
    netwatch = True
    no_ssl_certificate = False
    capsman_clients = True
    wireless = True
    ssl_certificate_verify = False
    pool = True
    interface = True
    monitor = True
    use_ssl = False
    capsman = False
    enabled = True
    dhcp_lease = True
    connections = True
    wireless_clients = True
    dhcp = True
    poe = False
    firewall = True
    hostname = 192.168.1.1
    username = redacted
    password = redacted
    port = 8728

[outdoor-ap]
    dhcp_lease = True
    dhcp = True
    route = True
    capsman = False
    connections = True
    interface = True
    monitor = True
    firewall = True
    netwatch = True
    pool = True
    capsman_clients = False
    wireless_clients = True
    use_comments_over_names = True
    no_ssl_certificate = False
    enabled = True
    wireless = True
    poe = False
    use_ssl = False
    ssl_certificate_verify = False
    hostname = 192.168.1.158
    username = redacted
    password = redacted
    port = 8728
```

Only the `outdoor-ap` has this issue.

**Comment by akpw** [comment]
@TrueWinter, thanks for details report. @M0r13n recently submitted [a fix](#39) that should take care of that, which now has been integrated into [the latest](https://pypi.org/project/mktxp/). So would you pls try it out?

Regarding the cause of the problem, looking at the last line of the Exception:
`labels={'channel': None, 'routerboard_name': 'outdoor-ap', 'routerboard_address': '192.168.1.158'}`,  looks like occasionally there is no channel from the outdoor-ap which is what was causing the exception.

**Comment by TrueWinter** [comment]
Looks like that fixed it, thanks.
